### DISCUSSION TITLE
ERG mode custom adjustment

### DISCUSSION THREAD

**Discussion by mrmichaelf** [dvmarinoff/Auuki#144] [General]
Hi @dvmarinoff,
Thanks for this awesome app! Question:
Is there a way in the code to set a custom algorithm for ERG mode adjustment?
For example, if I think my trainer's ERG mode is too reactive, I want to code a different hysteresis like algorithmic rules to when and by how the trainer should rebalance flywheel resistance to comply with ERG power settings.
Inspecting the repo, looks like we can set a power for ERG, and then the adjustment algorithm is a black box by trainer firmware.
Thanks,

**Comment by dvmarinoff** [comment]
Not sure I follow, but the best place to add some code affecting ERG is here: https://github.com/dvmarinoff/Flux/blob/master/src/ble/controllable.js#L67

> looks like we can set a power for ERG, and then the adjustment algorithm is a black box by trainer firmware

Yep, this App only deals with the differences in the multiple communication protocols.

**Reply by mrmichaelf** [reply]
Got it.
1. The use case is ERG interval, going up in rpm so the trainer increases flywheel resistance, then it's hard. you slow down, the trainer decreases etc. and the interval is not stable. I'm looking into overriding trainer default algo making those adjustments.
2. Question if it's even possible, I guess no...
3. What I'll try to do, is to modify mentioned function such it will receive initial interval ERG watt requirement, and then "internally" adjusting by reading actual wattage. So the adjustment won't be direct rather "new" ERG watt setting within the interval. Let's see how this will fit real time.
